src/CodeGen_HLS_Target.h

```diff
@@ -59,7 +59,11 @@ class CodeGen_HLS_Target {
         void visit(const For *op);
         void visit(const Allocate *op);
     private:
-        std::string guess_name(const std::string &name, uint32_t index);
+        /**
+         * Attempt to extract useful names from the arg name.
+         * If it fails, fall back to arg_%d as before.
+         */
+        static std::string get_arg_name(const std::string &name, uint32_t index);
```

Comment:
Make it a const member? E.g.
  static std::string get_arg_name(const std::string &name, uint32_t index) const;